clients/iscmove/iscmoveclient/client_assets_bag.go

```diff
@@ -25,33 +25,53 @@ func (c *Client) GetAssetsBagWithBalances(
 			ID:   *assetsBagID,
 			Size: uint64(len(fields.Data)),
 		},
-		Balances: make(iscmove.AssetsBagBalances),
+		Assets: *iscmove.NewEmptyAssets(),
 	}
 	for _, data := range fields.Data {
-		resGetObject, err := c.GetObject(ctx, iotaclient.GetObjectRequest{
-			ObjectID: &data.ObjectID,
-			Options:  &iotajsonrpc.IotaObjectDataOptions{ShowContent: true},
-		})
+		// for coins the "field name" is of type 0x1::ascii::String
```

Comment:
Is this really the case for only coins and nothing else?

Reply:
Yes, the key of the coins in the assets bag is the cointype converted to string, and for objects the key is the object ID.

:man_shrugging: